src/utils/extract_tenant_id.ts

```
@@ -0,0 +1,13 @@
+/**
```

Do not silently handle the failure here. Since the first part (index one) of the URL is expected to represent the tenant, verify it explicitly.
If the URL contains a valid tenant ID, proceed; otherwise, throw an exception instead of inserting an 'unknown' tenant ID into the database.

Note:
The tenant ID should ideally come from the authentication token, not from the URL or any other source.

---

very valid point. It is handled now. Thanks